pkg/controller/srcdst_controller.go

```diff
@@ -139,8 +182,38 @@ func GetInstanceIDFromProviderID(providerID string) (*string, error) {
 	// i-12345678 and i-12345678abcdef01
 	// TODO: Regex match?
 	if strings.Contains(instanceID, "/") || !strings.HasPrefix(instanceID, "i-") {
-		return nil, fmt.Errorf("Invalid format for AWS instanceID (%s)", instanceID)
+		return nil, fmt.Errorf("invalid format for AWS instanceID (%s)", instanceID)
 	}
 
 	return &instanceID, nil
 }
+
+func (c *Controller) getNodeObjByKey(key string) (nodeObj *v1.Node, err error) {
+	nodeItem, exists, err := c.nodeInformer.GetIndexer().GetByKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	if !exists {
+		return nil, fmt.Errorf("node object %s doesn't exist", key)
+	}
+
+	return nodeItem.(*v1.Node), nil
+}
+
+func (c *Controller) checkSrcDstAttributeEnabled(key string) (enabled bool, err error) {
+	defer c.workqueue.Done(key)
```

Comment:
Having this method call 'c.workqueue.Done' is a bit hard to follow.  Perhaps it can be placed before :
https://github.com/ottoyiu/k8s-ec2-srcdst/pull/16/files#diff-bff40c211a4e0c5aed5c55ab3cdbda41R10

and have the failure cases on 'processNextWorkItem' return early instead.